org.eclipse.xtext.xbase.tests/src/org/eclipse/xtext/xbase/tests/compiler/CompilerTests2.xtend

```diff
@@ -1856,4 +1856,16 @@ class CompilerTests2 extends AbstractOutputComparingCompilerTests {
 		''')
 	}
 
+	@Test def void test406762_NestedReturn() {
+		// this requires argument preparation
+		'''
+			{
+				return return 0
+			}
+		'''.compilesTo('''
+			return 0;
+			return error - couldn't compile nested return;
```

Comment:
I saw the same pattern in the XbaseCompiler. You're suggesting to generate that part in a Java comment?